Update Github issue templates to use issue forms

msulprizio opened the pull request:
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard

### Describe the update

Github issue template files are updated to utilize [issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). This involves using YAML configuration files instead of markdown. This should make issue submissions easier for users and will ensure users provide the required information.
    
Note that issue forms are not supported for pull requests currently. 
    
### Expected changes

This is a zero difference update to model output. It will only impact the look of submitting issues on Github once merged into the `main` branch.

### Related Github Issue(s)

- Closes #1831
- https://github.com/geoschem/HEMCO/pull/276
- https://github.com/geoschem/GCClassic/pull/58
- https://github.com/geoschem/GCHP/pull/415

msulprizio commented:
A temporary repository has been set up at https://github.com/msulprizio/test_issues_form/issues/new/choose for testing these new forms without merging into `main` here.

yantosca commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,88 @@
+    id: log-files
+    attributes:
+      label: Please attach any relevant configuration and log files.
+      description: Drag and drop files like the GEOS-Chem log file, geoschem_config.rc, and HEMCO_Config.rc here. You may need to rename files to end in .txt first.
```
Typo: `geoschem_config.rc` -> `geoschem_config.yml`

Also add `HISTORY.rc` here

msulprizio replied:
Fixed typo and added `HISTORY.rc` to the list of files to provided in commit https://github.com/geoschem/geos-chem/pull/2288/commits/e4a555f0d86ea247dfb19883ed04850304a322b4

yantosca commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,88 @@
+      options:
+        - Intel
+        - GNU
+        - PGI
```
You may want to say `PGI (GEOS-Chem in CESM only)`

@lizziel would know the other compilers used by CESM

yantosca replied:
Also would be nice to have a text box where they can type in the compiler version.

yantosca replied:
I would also list GNU first, so that they are in alphabetical order

msulprizio replied:
In commit https://github.com/geoschem/geos-chem/pull/2288/commits/9a12d4819a4235dce35a0b7267eaac0e33f2727a, I removed the compiler dropdown and instead provide a text box for users to specify the compiler and version used

<img width="487" alt="Screenshot 2024-05-21 at 2 18 07 PM" src="https://github.com/geoschem/geos-chem/assets/4805437/b0adae41-d6e0-48e2-ac8d-43ca1d119226">

lizziel replied:
We are referring CESM users to report issues at the CESM forum. I think we should limit the compilers listed to be for the offline model.

yantosca replied:
The text box is fine.  Thanks @msulprizio

yantosca reviewed (changes requested):
Thanks @msulprizio.  This looks great overall, I've just noted a few minor things.

msulprizio commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,88 @@
+    id: log-files
+    attributes:
+	label: Please attach any relevant configuration and log files.
+	description: Drag and drop files like the GEOS-Chem log file, geoschem_config.rc, and HEMCO_Config.rc here. You may need to rename files to end in .txt first.
```
`build_info/summarize_build` has now been added to the list of files to provided

yantosca replied:
One small detail, summarize_build is a script.  Maybe say "the output from running build_info/summarize_build" or something like that.  Sorry for the nitpick.

lizziel reviewed (changes requested):
This looks great for GC-Classic issues. I wonder if we should have a drop-down for what simulation, e.g. fullchem, and configuration, e.g. GC-Classic nested, they are using. Sometimes GCHP users create issues on GEOS-Chem repo but there is no place for them to say they are using GCHP.

yantosca reviewed (approved):
Thanks for the updates @msulprizio!  Good to merge, pending @lizziel's additional comments

msulprizio commented:
> I wonder if we should have a drop-down for what simulation, e.g. fullchem, and configuration, e.g. GC-Classic nested, they are using. Sometimes GCHP users create issues on GEOS-Chem repo but there is no place for them to say they are using GCHP.

@lizziel That's a good point! I added additional entries to the form for configuration, simulation type, resolution, and met field. 

<img width="876" alt="Screenshot 2024-05-21 at 3 44 41 PM" src="https://github.com/geoschem/geos-chem/assets/4805437/4fd2779c-e897-442f-8aa1-aa52171f7a70">

Let me know if you think anything else should be added/changed!

yantosca commented:
@lizziel @msulprizio, this looks great!

lizziel commented:
Looks good! A few more comments: 
- If using GCHP then users should also provide MPI library and version
- If we include CESM and WRF-GC then we should also include GEOS
- I'm hesitant to start down the path of asking for libraries and config files for GEOS-Chem in external models. GCHP run issues should be in the GCHP github, and GEOS-Chem in GEOS issues go to the GMAO github, and CESM issues go to the CESM user forum. I don't think we give WRF-GC help either. I wonder if we should add a note pointing people to those resources?

lizziel commented:
Thinking more about this, I can see GMAO developers making an issue (although write now they just ping me on their own github issues) and NCAR developers too if they have a question or something breaks. But I still wonder if we should list other resources somewhere such as GCHP GitHub for GCHP issues, or explain GEOS-Chem repo issues are for all GC-Classic issues plus science codebase issues if run in GCHP or other models.

msulprizio commented:
@lizziel I removed the options related to external models from the configuration question and added a note for users to check with the external model's support.
